components/purchase_modal/purchase_modal.tsx

```diff
@@ -64,26 +86,13 @@ export default class PurchaseModal extends React.PureComponent<Props, State> {
 
         this.state = {
             paymentInfoIsValid: false,
-            productPrice: 0,
             billingDetails: null,
             cardInputComplete: false,
             processing: false,
+            selectedProduct: findProductInDictionary(props.products!),
```

Comment:
If we don't expect `props.products` to be null then we should remove it from a possible type above in the type definition